Add text-input component

ThatGuySam opened the pull request:
Adds styled text input for email/password field as well as an option for a bordered field 

Normal Field
`<TextInput name='email' placeholder='EMAIL' onChange={(e) => console.log('Field Changed', e)} />`

Bordered Field
`<TextInput name='first_name' placeholder='FIRST NAME' className='b--black-30' />`

Fixes #9

michael-martin-al commented on src/components/text-input/index.js:
```diff
@@ -0,0 +1,28 @@
+			type='text'
+			{...props}
+			name={props.name}
+			defaultValue={props.defaultValue}
```
@ThatGuySam Since you're passing through the entire `props` object, is there any benefit to setting them explicitly here? Perhaps you get the benefit of explicitly declaring the component contract in the propTypes. Thoughts?

ThatGuySam replied:
Not really. Code Climate was giving me a warning about not using my props that I declared so I added them in. The only only real benefit is that Code Climate will be happy. 
I can take them out if you prefer.

michael-martin-al replied:
No, it's fine. Leave them. That way, anyone who uses this component will see the expectations spelled out in propTypes.

michael-martin-al commented on src/components/text-input/index.js:
```diff
@@ -0,0 +1,28 @@
+TextInput.defaultProps = {
+	defaultValue: '',
+	// Clear inset shadow by default
+	className: 'b--transparent'
```
I'm trying to wrap my head around what you did here with the inset shadow. So, if no `className` is defined, we get the default behavior of `b--transparent`. But, if I do provide a `className` prop when using the component, it defaults to showing the inset shadow. At first glance, I think we should either...

1. Put `b--transparent` in the list of classes on line 11 so that it's always defined or
2. Make a separate prop for inset shadow 

I think the way it's coded now is confusing for the consumer. If I were to write this:

```
<TextInput className="red" />
```

Now, the inset shadow is back just because I changed my text color. I may be misinterpreting what you wrote. Thoughts?

ThatGuySam replied:
There are 2 different types of fields in the mockups: None-bordered and bordered. 

I was having a hard time thinking of an intuitive way to use this component for both because you have to set a border class otherwise you'll get the inset shadow. 

But because of the class priority in tachyons if you set `b--transparent` you can't override it with `b--black-30` in the same class list. 

The idea is that by default it will be in the style of the one you specified(EMAIL/PASSWORD) 
`<TextInput />`
and when we get to the bordered fields(FIRST NAME/LAST NAME) we just add the class to set the border
`<TextInput className='b--black-30' />`
The drawback of this strategy is if you want to add any class you also have to add a border color class otherwise you'll get the inset border. 

One idea could be to add `border: none;` to .input-reset and put it at a lower priority than the border class helpers but custom css is a last resort. 

Also I could just make a bordered TextField component.

ThatGuySam commented:
Okay, I added a function to just look for `b--` in the class list and add `b--transparent` when it's not there. This mean if you add the class `red` there won't be a inset border

ThatGuySam commented:
Whoops, just saw that part about User Registration page, It's there now. 
https://deploy-preview-19--img-src.netlify.com/user-registration

michael-martin-al commented:
@ThatGuySam sorry for the hassle. I think you were thinking about the bigger picture when I wasn't. This looks good now. When I saw your two examples in context it clicked. Are you good with me merging this now?

ThatGuySam commented:
My bad, by default the field has a white bg and is designed to be in front of a off-white bg like in the EMAIL/PASSWORD example. 

<img width="312" alt="screenshot 2018-05-07 10 55 58" src="https://user-images.githubusercontent.com/2830379/39711673-f621a256-51e5-11e8-8884-7210fb395fb2.png">

I updated the example to show that more clearly. 
Let me if you'd rather me change the default. 

I changed the padding from `pa3` to `pv3 ph4` to better match the mockup.

ThatGuySam commented:
Ha! Thanks for the credit! Merge away!!!
